clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/UbamTests.scala

```diff
@@ -313,6 +313,63 @@ trait UbamTests { self: BaseIntegrationSpec =>
     }
   }
 
+  it should "handle querying ubams by aggregated-by" in {
+    val flowcellBarcode = "barcode2"
+    val lane = 2
+    val location = Location.GCP
+    val project = "testProject" + randomId
+
+    val libraries = Seq.fill(3)("library" + randomId)
+    val samples = {
+      val sameId = "testSample" + randomId
+      Seq(sameId, sameId, "testSample" + randomId)
+    }
+    val researchProjectIds = Seq.fill(3)("rpId" + randomId)
+
+    val aggregations = Seq(
+      AggregatedBy.Squid,
+      AggregatedBy.Squid,
+      AggregatedBy.RP
+    )
+
+    val upserts = Future.sequence {
+      libraries zip samples zip researchProjectIds zip aggregations map {
```

Comment:
Since we aren't asserting anything about the libraries, we could not include them and use zip3

Reply:
we need different libraries or else all the keys would be the same